packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -806,8 +806,12 @@ abstract class FlutterCommand extends Command<void> {
 
     if (shouldRunPub) {
       await pub.get(context: PubContext.getVerifyContext(name));
+      // All done updating dependencies. Release the cache lock.
+      Cache.releaseLock();
```

Comment:
Moved `releaseLock` so it can run before `ensureReadyForPlatformSpecificTooling`.